allennlp/data/dataset_readers/text_classification_json.py

```diff
@@ -0,0 +1,120 @@
+from typing import Dict, List
+import logging
+import json
+from overrides import overrides
+from allennlp.common.file_utils import cached_path
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.fields import LabelField, TextField, Field, ListField
+from allennlp.data.instance import Instance
+from allennlp.data.token_indexers import TokenIndexer, SingleIdTokenIndexer
+from allennlp.data.tokenizers import Tokenizer, WordTokenizer
+from allennlp.data.tokenizers.sentence_splitter import SpacySentenceSplitter
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@DatasetReader.register("text_classification_json")
+class TextClassificationJSONReader(DatasetReader):
```

Comment:
Nit: it should be `Json`, not `JSON`(see [here](https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)).